cv32e40s/tests/programs/custom/zc_test/zc_test.c

```diff
@@ -0,0 +1,214 @@
+/*
+**
+** Copyright 2020 OpenHW Group
+**
+** Licensed under the Solderpad Hardware Licence, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+**     https://solderpad.org/licenses/
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+*******************************************************************************
+**
+** Directed test for Zc extension   exersizes instructions and scenarios
+**                                  not likely to becovered by randomly
+**                                  generated tests.
+**
+*******************************************************************************
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "zc_test.h"
+
+
+//       Program Functions                //
+
+//trigger interrupt
+void vp_assert_irq(uint32_t mask, uint32_t cycle_delay) {
+    *TIMER_REG_ADDR = mask;
+    *TIMER_VAL_ADDR = 1 + cycle_delay;
+}
+
+uint32_t vp_random_num(uint32_t upper_bound, uint32_t lower_bound) {
+    uint32_t random_num = *((volatile uint32_t *) CV_VP_RANDOM_NUM_BASE);
+    uint32_t num = (random_num  % (upper_bound - lower_bound + 1)) + lower_bound;
```

Comment:
I would like an assertion of some kind to stop compilation and give a meaningful error message that the combination of upper_bound = 0xffff_ffff and lower_bound = 0x0 is not supported, or a fix to the potential overflow leading to a divide-by-zero error here, otherwise future modifications might accidentally modify the input parameters to this function such that things break.
The compiler detects this overflow condition and inserts a trap to avoid performing illegal arithmetic, and this can be somewhat complicated to debug.

Reply:
Fix that stops execution will be added.